### DISCUSSION TITLE
如何每章设置一个参考文献？

### DISCUSSION THREAD

**Discussion by zhoujian9410** [nju-lug/NJUThesis#75] [Q&A] [answered]
参考文献较多，因此想分章节进行参考文献引用，想请教一下现在的模板能否实现这个目标呢？能的话应该如何设置？多谢各位大佬

**Comment by zhoujian9410** [comment, marked as answer]
我好像找到了一个方法，如下用需要独立生成参考文献的章节用{refsection}包裹即可，来源：[Separate bibliographies per chapter?
](https://texfaq.org/FAQ-chapbib)

```
\begin{refsection}
\chapter{First chapter}
\section{Foo}
Some text \cite{this}
with citations \cite{that}.
\printbibliography
\end{refsection}
```
但是，我还有一个问题，\printbibliography命令生成的参考文献是和chapter一个目录层次的，能否有办法降低参考文献的目录层次到section级别？Latex小白，感谢大佬们赐教

**Comment by atxy-blip** [comment]
`refsection` 环境确实是正解，恭喜！

> 降低参考文献的目录层次到section级别

加上可选参数即可，根据你的喜好二选一：

```
\printbibliography[heading=subbibliography] % 不插入目录的 section
\printbibliography[heading=subbibintoc] % 插入目录的 section
```

具体用法见[相关宏包的使用手册](http://mirrors.ctan.org/macros/latex/contrib/biblatex/doc/biblatex.pdf)：

![image](https://user-images.githubusercontent.com/63113546/153758337-f282375c-4b23-4ac3-8628-29c94529adb0.png)

**Reply by zhoujian9410** [reply]
嗯嗯，现在可以了，感谢指导以及分享的书目，学习了。
另外直接用
```
\printbibliography[heading=subbibintoc]

```
会出现“Undefined control sequence. \printbibliography[heading=subbibintoc]”错误（Texlive 2021+Texstudio），加上title可以避开这个问题（见下），我不知道原因，先分享在这里。
```
\printbibliography[heading=subbibintoc,title={参考文献}]
```

**Comment by atxy-blip** [comment]
新版本（v0.15.0）加了一个选项，发布后可以在导言区加上这样一个设置
```
\njusetup {
    bib = { option = { refsection = chapter } }
  }
```
随后在每章后直接加 `\printbibliography` 就好。

**Reply by zhoujian9410** [reply]
太强了，辛苦，感谢&#x1F64F;
